NOTES.rst

```diff
@@ -21,7 +21,27 @@ contains the same SingleByteCharSetProbers.
 SingleByteCharSetProber
 -----------------------
 A CharSetProber that is used for detecting single-byte encodings by using
-a "precedence matrix" (i.e., a character bigram model).
+a "precedence matrix" (i.e., a character bigram model).  The weird thing about
+this precedence matrix is that it's not actually based on all sequences of
+characters, but rather just the 64 most frequent letters, numbers, and control
+characters.  (We should probably have control characters not count here like in
+https://github.com/BYVoid/uchardet/commit/55b4f23971db61c9ed93be6630c79a50bda9b.)
+To look things up in the language model, we actually look up by "frequency order"
+(as in CharDistributionAnalysis), so that we can use one language model for multiple encodings.
+
+Furthermore, when calculating the confidence what we actually are counting is
+the number of sequences we've seen of different likelihoods:
+
+-  positive = in the 512 most frequent sequences
+-  likely = in the 1024 most frequent sequences
+-  unlikely = occurred at least 3 times in training data
+-  negative = did not occur at least 3 times in training data
+
+We should probably allow tweaking these thresholds when training models, as 64
```

Comment:
These notes are slightly out of date, as we now use the length of the alphabet for the language instead of 64 here.